src/pip/_internal/commands/freeze.py

```diff
@@ -80,9 +78,12 @@ def run(self, options, args):
         # type: (Values, List[str]) -> int
         format_control = FormatControl(set(), set())
         wheel_cache = WheelCache(options.cache_dir, format_control)
-        skip = set(stdlib_pkgs)
-        if not options.freeze_all:
-            skip.update(DEV_PKGS)
+        if options.freeze_all_used:
+            deprecated(
+                "--all is now the default behavior and does nothing",
+                None,
+                gone_in="23.3",
```

Comment:
Why use Oct 2023 instead of July 2021? :P

```suggestion
                gone_in="21.2",
```

Reply:
```suggestion
                gone_in="23.0",
```